sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -587,15 +673,29 @@ class reduction_impl : private reduction_impl_base {
   }
 
   /// Constructs a new temporary buffer to hold partial sums and returns
-  /// the accessor that that buffer.
-  template <bool IsOneWG>
-  std::enable_if_t<!IsOneWG, accessor_type>
+  /// the accessor that that buffer. Non-placeholder case.
+  template <bool IsOneWG, access::placeholder _IsPlaceholder = IsPlaceholder>
+  std::enable_if_t<!IsOneWG && _IsPlaceholder == access::placeholder::false_t,
+                   accessor_type>
   getWriteMemForPartialReds(size_t Size, handler &CGH) {
     MOutBufPtr = std::make_shared<buffer<T, buffer_dim>>(range<1>(Size));
     CGH.addReduction(MOutBufPtr);
     return accessor_type(*MOutBufPtr, CGH);
   }
 
+  /// Constructs a new temporary buffer to hold partial sums and returns
+  /// the accessor that that buffer. Placeholder case.
```

Comment:
```suggestion
  /// the accessor for that buffer. Placeholder case.
```

Reply:
Fixed in 0cdf646139cb6f27a28f985cd123eb96ebd540be